api/Dockerfile

```diff
@@ -23,21 +24,29 @@ RUN adduser \
 WORKDIR /app
 
 # First we handle fetching and building our dependencies
-# We need a stub entry point for that
+# This rarely changes so is done once and for all
+# Cargo needs a stub entry point otherwise it fails
 COPY Cargo.toml Cargo.lock ./
 COPY ./src/stub.rs ./src/main.rs
```

Comment:
There is no need for `stub.rs`, `main.rs` works as well as long as you do not build later before copying the other files. Anyways we build twice now which is quite overkill.

Reply:
We can not use the original `main.rs` because it would cause a rebuild of the dependencies.

I would prefer if cargo was able to build dependencies without the `stub.rs`, but it was not the case when I tested.

The project is built only once:
- the dependencies are built here, which takes ~2min;
- later our api code is built, which takes ~5s.

Reply:
You can do `RUN echo "fn main() { }" > src/main.rs `